qcodes/instrument_drivers/devices.py

```diff
@@ -1,6 +1,190 @@
+
+        # Set the name
+        if name:
+            self.name = name
+        else:
+            if self.role == Role.DIVISION:
+                self.name = "{}_attenuated".format(self._wrapper_param.name)
+            elif self.role == Role.GAIN:
+                self.name = "{}_amplified".format(self._wrapper_param.name)
+
+        # Set label
+        if label:
+            self.label = label
+        elif name:
+            self.label = name
+        else:
+            if self.role == Role.DIVISION:
+                self.label = "{}_attenuated".format(self._wrapper_param.label)
+            elif self.role == Role.GAIN:
+                self.label = "{}_amplified".format(self._wrapper_param.label)
+
+        # Set the unit
+        if unit:
+            self.unit = unit
+        else:
+            self.unit = self._wrapper_param.unit
+
+        super().__init__(
+            name=self.name,
+            instrument=getattr(self._wrapper_param, "_instrument", None),
+            label=self.label,
+            unit=self.unit,
+            metadata=self._wrapper_param.metadata)
+
+        # extend metadata
+        self._meta_attrs.extend(["division"])
+        self._meta_attrs.extend(["gain"])
+
+    # Internal handling of the multiplier
+    # can be either a Parameter or a scalar
+    @property
+    def _multiplier(self):
+        return self._multiplier_parameter
+
+    @_multiplier.setter
+    def _multiplier(self, multiplier: Union[int, float, Parameter]):
+        if isinstance(multiplier, Parameter):
+            self._multiplier_parameter = multiplier
+        else:
+            self._multiplier_parameter = ManualParameter(
+                'multiplier', initial_value=multiplier)
+
+    # Division of the scaler
+    @property
+    def division(self):
+        if self.role == Role.DIVISION:
+            return self._multiplier()
+        elif self.role == Role.GAIN:
+            return 1 / self._multiplier()
+
+    @division.setter
+    def division(self, division: Union[int, float, Parameter]):
+        self.role = Role.DIVISION
+        self._multiplier = division
+
+    # Gain of the scaler
+    @property
+    def gain(self):
+        if self.role == Role.GAIN:
+            return self._multiplier()
+        elif self.role == Role.DIVISION:
+            return 1 / self._multiplier()
+
+    @gain.setter
+    def gain(self, gain: Union[int, float, Parameter]):
+        self.role = Role.GAIN
+        self._multiplier = gain
+
+    # Getter and setter for the real value
+    def get_raw(self) -> Union[int, float]:
+        """
+        Returns:
+            number: value at which was set at the sample
+        """
+        if self.role == Role.GAIN:
+            value = self._wrapper_param() * self._multiplier()
+        elif self.role == Role.DIVISION:
+            value = self._wrapper_param() / self._multiplier()
+
+        self._save_val(value)
+        return value
+
+    def get_instrument_value(self) -> Union[int, float]:
```

Comment:
Why is it called *instrument* value? What is returned is the value of the wrapped parameter, so I expect that the name is smth like “get_unscaled_value” or better “get_wrapped_parameter_value”.